libraries/rush-lib/src/logic/operations/BuildTimePlugin.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  ICreateOperationsContext,
+  IPhasedCommandPlugin,
+  PhasedCommandHooks
+} from '../../pluginFramework/PhasedCommandHooks';
+import { IExecutionResult } from './IOperationExecutionResult';
+import { OperationStatus } from './OperationStatus';
+
+const PLUGIN_NAME: 'BuildTimePlugin' = 'BuildTimePlugin';
+
+interface ITimelineRecord {
+  startTime: number;
+  endTime: number;
+  durationString: string;
+  name: string;
+  status: OperationStatus;
+}
+
+export interface IBuildTimeRecord {
+  project: string;
+  buildTime: number;
+  status: OperationStatus;
+}
+
+export class BuildTimePlugin implements IPhasedCommandPlugin {
+  private _buildTimes: IBuildTimeRecord[];
+
+  public constructor() {
+    this._buildTimes = [];
+  }
+
+  public apply(hooks: PhasedCommandHooks): IBuildTimeRecord[] {
+    hooks.afterExecuteOperations.tap(
+      PLUGIN_NAME,
+      (result: IExecutionResult, context: ICreateOperationsContext): void => {
+        this._buildTimes = _setBuildTimes(result);
+      }
+    );
+    return this._buildTimes;
+  }
+}
+
+export function _setBuildTimes(result: IExecutionResult): IBuildTimeRecord[] {
+  const data: ITimelineRecord[] = [];
+  const buildTimes: IBuildTimeRecord[] = [];
+
+  for (const [operation, operationResult] of result.operationResults) {
+    if (operation.runner?.silent) {
+      continue;
+    }
+
+    const { stopwatch } = operationResult;
+
+    const { startTime, endTime } = stopwatch;
+
+    if (startTime && endTime) {
+      const { duration } = stopwatch;
+      const durationString: string = duration.toFixed(2);
```

Comment:
Why convert to a string and then back into a number? Seems like it would make more sense to record the full number and only round at display time.

Reply:
I borrowed code from ConsoleTimelinePlugin to do the time calculations and reused the `ITimelineRecord` interface. In hindsight I don't need that much structure in my interface, especially the string type for durationString. I will refactor this to be a smaller interface with a number as the type for duration. Good catch.